Makefile

```diff
@@ -221,10 +221,13 @@ catalog-push: ## Push a catalog image.
 	$(MAKE) docker-push IMG=$(CATALOG_IMG)
 
 .PHONY: test-sanity
-test-sanity: tidy vet fmt
+test-sanity: tidy git-diff vet fmt
 
 .PHONY: tidy
 tidy:
 	go mod tidy
+
+.PHONY: git-diff
+git-diff:
```

Comment:
What's the reason for splitting this out? This is mainly used in CI (if we aren't we should be). So the git diff --exit-code should still be a part of the target.

Reply:
+1

Reply:
IMO it makes sense to separate tidy and git diff into 2 targets.   I took care to call both from `sanity-test` (the only place where tidy was being called).

To me it felt out of place to do a git diff inside a target that's supposed to do go mod tidy.

Originally did that because I wanted to run `tidy` as part of the build & the git diff failed.  but the result was a slightly bigger image at ±2.3gb so went with `-a` eventually which brings size down to 2.1gb.

Reply:
That tidy target really is just meant for CI consumption IMO. I don't feel strongly about it... But it might end up being more verbose if we need to run 'make tidy git-diff' and 'make fmt git-diff' and 'make vet git-diff'.

Reply:
And... I don't think tidy should be automatic. Just like the discussion we had about make bundle. CI should inform the change. But that shouldn't be something automatic.

Reply:
But I don't feel strongly against changing it back to what it was.

Reply:
Maybe then leave it as is since you aren't using it for what you intended and we can have another discussion about it?

Reply:
fair enough.  changing back.